src/tests/Common/XUnitLogChecker/XUnitLogChecker.cs

```diff
@@ -30,41 +30,116 @@ public TagResult(string matchValue, TagCategory matchCategory)
+    }
+
     private enum TagCategory { OPENING, CLOSING }
 
     private const int SUCCESS = 0;
     private const int MISSING_ARGS = -1;
     private const int FAILURE = -2;
 
+    private static LogCheckerConfigParameters s_configuration;
+
     static int Main(string[] args)
     {
-        if (args.Length < 2)
+        s_configuration = new LogCheckerConfigParameters();
+
+        // We start with a 'FAILURE' exit code because if something goes wrong
+        // later on and the log checker fails, we want to know it.
+        int exitCode = FAILURE;
+
+        ParseCommandLineArgs(args);
+
+        // In CoreCLR tests, we record each test's results in an XML file, which
+        // is stored in the same directory as said test's script. So, the
+        // XUnitLogChecker looks there for said results log to do its job of
+        // fixing it if necessary. Afterwards, it checks if there are any dumps
+        // to read and display. This is the log checker's full run.
+        //
+        // In the Libraries tests however, the test logs are stored and created
+        // in a different way, currently outside the scope of the XUnitLogChecker,
+        // and therefore only the dump checking functionality is required in
+        // those cases, so we skip directly to that step.
+
+        if (s_configuration.HasResultsPath() && s_configuration.HasTestWrapper())
         {
-            WriteLineTimestamp("The path to the log file and"
-                               + " the name of the wrapper are required for an"
-                               + " accurate check and fixing.");
-            return MISSING_ARGS;
+            WriteLineTimestamp("The full run will be done.");
+            exitCode = DoFullRun();
         }
+        else
+        {
+            WriteLineTimestamp("Only dumps checking will be done.");
+            DoDumpsCheck(string.Empty);
+            exitCode = SUCCESS;
+        }
+
+        return exitCode;
+    }
 
-        // Creating variables for code clarity and ease of understanding.
+    private static void ParseCommandLineArgs(string[] args)
+    {
+        // The command-line arguments will always come in pairs of '--flag value'.
+        // Hence, we process two elements of the args array at a time.
 
-        string resultsDir = args[0];
-        string wrapperName = args[1];
+        for (int i = 0; i < args.Length; i+=2)
+        {
+            string nextOption = args[i];
+            string nextValue = (i+1) < args.Length ? args[i+1] : string.Empty;
+
+            switch (nextOption)
+            {
+                case "--results-path":
+                    s_configuration.ResultsPath = nextValue;
+                    break;
+
+                case "--test-wrapper":
+                    s_configuration.TestWrapperName = nextValue;
+                    break;
+
+                case "--dumps-path":
+                    s_configuration.DumpsPath = nextValue;
+                    break;
```

Comment:
For the 3 cases, if `nextValue` is `string.Empty`, I think we should throw an exception indicating that "a value for <case name> was not provided". This is especially necessary since `DoFullRun`'s first few lines assume the TestWrapperName` and `ResultsPath` have a valid non-empty value.

Reply:
We are already checking for the arguments to not be empty where needed so this is not necessary.

Reply:
If a value isn't passed, the forloop is going to misbehave.

`XUnitLogChecker.exe --results-path --test-wrapper --dumps-path`

Reply:
This is not meant to be a tool used for general consumption. This is specifically for our test infrastructure with specific calls that should be made correctly in the first place (i.e. that case you describe should never happen)